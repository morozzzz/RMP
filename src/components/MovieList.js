import React from 'react';
import FilterBar from './FilterBar';
import GridList from './GridList';

