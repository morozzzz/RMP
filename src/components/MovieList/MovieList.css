.movie-list {
    display: flex;
    flex-direction: column;
}

.movie-list__message {
    font-size: 3em;
    font-weight: bold;
    flex-grow: 1;
    display: flex;
    justify-content: center;
    align-items: center;
}