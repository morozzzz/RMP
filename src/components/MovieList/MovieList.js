import React from 'react';
import GridList from '../GridList/GridList';
import './MovieList.css';

const MovieList = ({ movies, onItemClick }) => (
    <div className="movie-list">
        {
            movies.length
                ? <GridList data={movies} onItemClick={onItemClick} />
                : <div className="movie-list__message">No films found</div>
        }
    </div>
);

export default MovieList;
