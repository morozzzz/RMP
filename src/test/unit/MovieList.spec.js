import React from 'react';
import { shallow } from 'enzyme';
import MovieList from '../../components/MovieList/MovieList';

const moviesMock = [
    {
        id: 333339,
        title: 'Ready Player One',
        tagline: 'A better reality awaits.',
        vote_average: 8.1,
        vote_count: 617,
        release_date: '2018-03-28',
        poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
        overview: 'When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.',
        budget: 175000000,
        revenue: 0,
        genres: [
            'Adventure',
            'Science Fiction',
            'Action',
        ],
        runtime: 140,
    },
    {
        id: 338970,
        title: 'Tomb Raider',
        tagline: 'Her legend begins',
        vote_average: 6.2,
        vote_count: 817,
        release_date: '2018-03-08',
        poster_path: 'https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg',
        overview: 'Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.',
        budget: 94000000,
        revenue: 126025000,
        genres: [
            'Action',
            'Adventure',
        ],
        runtime: 118,
    },
];

jest.mock('../../components/GridList/GridList', () => () => <div id="GridList" />);

describe('MovieList', () => {
    it('should be rendered correctly', () => {
        const wrapper = shallow(<MovieList movies={moviesMock} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('container with message should be shown if there are no movies', () => {
        const wrapper = shallow(<MovieList movies={[]} />);

        expect(wrapper.find('.movie-list__message')).toBeDefined();
    });
});
