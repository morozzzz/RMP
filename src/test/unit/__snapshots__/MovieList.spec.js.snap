// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should be rendered correctly 1`] = `
<div
  className="movie-list"
>
  <Component
    data={
      Array [
        Object {
          "budget": 175000000,
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "revenue": 0,
          "runtime": 140,
          "tagline": "A better reality awaits.",
          "title": "Ready Player One",
          "vote_average": 8.1,
          "vote_count": 617,
        },
        Object {
          "budget": 94000000,
          "genres": Array [
            "Action",
            "Adventure",
          ],
          "id": 338970,
          "overview": "Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.",
          "poster_path": "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
          "release_date": "2018-03-08",
          "revenue": 126025000,
          "runtime": 118,
          "tagline": "Her legend begins",
          "title": "Tomb Raider",
          "vote_average": 6.2,
          "vote_count": 817,
        },
      ]
    }
  />
</div>
`;
